src/application/tester.go

```diff
@@ -17,7 +17,12 @@ func TestApp(appContext types.AppContext, logger *util.Logger, mode composebuild
 	}
 
 	numFailed := 0
+	locations := []string{}
 	for serviceName, serviceContext := range serviceContexts {
+		if util.DoesStringArrayContain(locations, serviceContext.Location) {
+			continue
+		}
+		locations = append(locations, serviceContext.Location)
```

Comment:
This is preventing from running the same tests twice when a service is reused. Wasn't exactly certain how to test this

Reply:
This seems simple and sound enough. If you want, you could extract a `forServiceDirs` method like we have `forEachService` in the `types/app-config.go` that just loops through service locations and maybe test that method. We might be using something like that in `exo clean` as well you said?

Reply:
That sounds good